src/libraries/System.Text.RegularExpressions/tests/Regex.Count.Tests.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexCountTests
+    {
+        [Theory]
+        [MemberData(nameof(Count_ReturnsExpectedCount_TestData))]
+        public async Task Count_ReturnsExpectedCount(RegexEngine engine, string pattern, string input, RegexOptions options, int expectedCount)
+        {
+            Regex r = await RegexHelpers.GetRegexAsync(engine, pattern, options);
+            Assert.Equal(expectedCount, r.Count(input));
+            Assert.Equal(r.Count(input), r.Matches(input).Count);
+
+            if (options == RegexOptions.None && engine == RegexEngine.Interpreter)
+            {
+                Assert.Equal(expectedCount, Regex.Count(input, pattern));
```

Comment:
why's this special?

Reply:
No options are accepted so it always uses the interpreter.